Refactor account documentation into sections (#1523)
0xMiden/miden-base: docs/src/account/code.md

- **Mutable access:** Only the `Account`'s own functions can modify its storage and vault. All state changes — such as updating storage slots or transferring assets — must occur through these functions.
- **Function commitment:** Each function can be called by its [MAST](https://0xMiden.github.io/miden-vm/user_docs/assembly/main.html) root. The root represents the underlying code tree as a 32-byte commitment. This ensures integrity which means a function's behavior cannot change without changing the MAST root.
- **Asset creation:** Faucet `Accounts` can create assets.

An account's code is typically the result of merging multiple [account components](./component.md).

## Authentication

Authenticating a transaction, and therefore the changes to the account, is done with an _authentication procedure_. Every account's code must provide exactly one authentication procedure. It is automatically called during the transaction epilogue, i.e. after all note scripts and the transaction script have been executed.

> Such authentication procedures typically inspect the transaction and then decide whether a signature is required to authenticate the changes. They do this by:

Reviewer: ```suggestion
Such an authentication procedure typically inspects the transaction and then decides whether a signature is required to authenticate the changes. It does this by:
```

# Account Code

> [!Note]
> A collection of functions defining the `Account`'s programmable interface.

Every Miden `Account` is essentially a smart contract. The `Code` defines the account's functions, which can be invoked through both [Note scripts](../note.md#script) and [transaction scripts](../transaction.md#inputs). Key characteristics include:

- **Mutable access:** Only the `Account`'s own functions can modify its storage and vault. All state changes — such as updating storage slots or transferring assets — must occur through these functions.
- **Function commitment:** Each function can be called by its [MAST](https://0xMiden.github.io/miden-vm/user_docs/assembly/main.html) root. The root represents the underlying code tree as a 32-byte commitment. This ensures integrity which means a function's behavior cannot change without changing the MAST root.

> - **Note creation:** `Account` functions can create new notes.

Author: This does make it sound like **only** account's can create notes, so I'll remove it.
